simple_salesforce/api.py

```diff
@@ -565,12 +571,8 @@ def _call_salesforce(self, method, url, **kwargs):
 
         Returns a `requests.result` object.
         """
-        headers = {
-            'Content-Type': 'application/json',
-            'Authorization': 'Bearer ' + self.session_id,
-            'X-PrettyPrint': '1'
-        }
-        result = self.request.request(method, url, headers=headers, **kwargs)
+
+        result = self.request.request(method, url, headers=self.headers, **kwargs)
```

Comment:
if self.headers: # use them, 
otherwise: use old.

In case someone was using references to Tasks or Leads for example.